### DISCUSSION TITLE
how to built the latest?

### DISCUSSION THREAD

**Discussion by PatLabGit** [lbalazscs/Pixelitor#360] [Q&A]
I downloaded the Linux / x64 tar.gz  openjdk 22 at https://jdk.java.net/22/ 
I uncompressed the archive, now what do I do with this directory? 
I'm on Ubuntu-MATE 20.04

![screenshot_20240406-155014](https://github.com/lbalazscs/Pixelitor/assets/47497693/ecd77687-aac3-469e-b74a-fb8160f3a1a8)

**Comment by lbalazscs** [comment]
In order to run the latest version of Pixelitor on Linux, you need two things: (1) a Java runtime and (2) a jar file that is created from Pixelitor's latest source code. You've already downloaded the latest Java runtime. This is a precompiled distribution, so there's no need to build it. However, you still need to have the latest jar file. The easiest way to get that is described here: https://github.com/lbalazscs/Pixelitor/wiki/Downloading-the-latest-development-build

Once you have that, you can run it with the command
```/home/patrice/Desktop/jdk-22/bin/java -jar /path/to/the/pixelitor_jar_file.jar```

Of course you can put this command into a shell script or into a desktop shortcut. 
Let me know if you have any questions!
Here's how the new French tips of the day should look in the newest version:
![french](https://github.com/lbalazscs/Pixelitor/assets/9939454/ef0d7f01-bc5e-4b67-b3f7-2c696f4c822a)

**Reply by PatLabGit** [reply]
Yesss!, it's working with >  /home/patrice/Desktop/jdk-22/bin/java -jar /path/to/the/pixelitor_jar_file.jar

This is how it looks on Linux (MATE Desktop)
![screenshot_20240407-141441](https://github.com/lbalazscs/Pixelitor/assets/47497693/9690f97d-566d-4fcc-ac92-0f26d351db57)

Now that I can see how it is shown, for example on my screenshot, I should have use the "subjonctif" (que / qu'), it would have been better... also first letter upper case in the listing...

Now to not always do a pull request, when not necessary, can you explain me how can I have that kind of page > https://github.com/lbalazscs/Pixelitor/actions/workflows/build.yml for me to try Pixelitor before to do a pull request? 

Actually my page is empty > https://github.com/PatLabGit/Pixelitor/actions/workflows/build.yml I cannot try Pixelitor before a pull request, I don't know which workflow I should use, etc...

**Reply by lbalazscs** [reply]
I don't know why is your build page empty. Maybe you need to enable the running of github action workflows. But anyway, if you want to contribute regularly I think it's better if you install Maven and create the jar file locally as described in https://github.com/lbalazscs/Pixelitor/blob/master/README.md (in "Building the Pixelitor jar file from the command line"). If you google "ubuntu maven install", you get a lot of tutorials. Let me know if you have questions or difficulties!

**Comment by lbalazscs** [comment]
Note that I recommend getting a Java runtime from https://adoptium.net because the last time I tried the one from jdk.java.net, it didn't had some of the translation files that are not Pixelitor-specific, but they are useful. It's a good idea to get Java 22 (instead of the "latest LTS", which currently is Java 21), because Pixelitor might require Java 22 soon (but the current version only requires Java 21)

**Reply by PatLabGit** [reply]
`I recommend getting a Java runtime from https://adoptium.net/`
That's the one I'm using now with terminal, :+1:
